console/console-init/ui/src/modules/iot-device-detail/components/CredentialsView/SecretsView/SecretsView.tsx

```diff
@@ -68,20 +71,26 @@ const SecretRow: React.FC<ISecretRowProps> = ({
 
   const renderGridItemValue = (value: string, key: string) => {
     if (key === "pwd-hash") {
-      return (
-        <Button
-          id="secrets-view-change-password-button"
-          variant="link"
-          icon={<EditAltIcon />}
-          className={classNames([
-            styles.c_button_PaddingLeft,
-            styles.c_button_PaddingBottom
-          ])}
-          onClick={onClickChangePassword}
-        >
-          Change password
-        </Button>
-      );
+      if (enableActions) {
```

Comment:
small change here, we need both check otherwise change password link will show for all credentuals type that will wrong.
if(key === "pwd-hash" && enableActions)